blocks/main_loop.py

```diff
@@ -231,6 +231,12 @@ def _run_iteration(self):
         return True
 
     def _run_extensions(self, method_name, *args):
+        class Callback(str):
+            def __eq__(self, other):
+                if other not in ['before_batch', 'after_batch']:
+                    raise ValueError("I'm being compared to something \
+                                      I don't know!")
```

Comment:
Thanks for your comment! I understood the rest part except this one. If other is a known callback. It should look like this: Callback(method_name),  right? so now I have something like Callback(Callback(method_name)), after having line 235/236, I need another "if". However, what's the meaning of str(self)? why not use self directly?